pkg/util/azureclient/azureclient.go

```diff
@@ -0,0 +1,75 @@
+package azureclient
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/marketplaceordering/mgmt/marketplaceordering"
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	azmarketplaceordering "github.com/Azure/azure-sdk-for-go/services/marketplaceordering/mgmt/2015-06-01/marketplaceordering"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2018-02-01/storage"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/openshift/openshift-azure/pkg/api"
+)
+
+// AzureClients holds all the used clients used by openshift-azure
+type AzureClients struct {
```

Comment:
Doing this for our plugin code is probably fine since all plugins are used by a single codebase but I have resisted the urge for having a single struct that holds all of the clients we use. A single import of this in `sync` is going to pull with it a lot of unnecessary code.

Reply:
@kargakis it was not my intention to use this in the sync code, just where it made sense.